databases/src/connect/DatabaseNameStep.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from 'vscode-azureextensionui';
+import { IConnectDBWizardContext } from './IConnectDBWizardContext';
+import { nonNullProp } from '../utils/nonNull';
+import { createCosmosDBClient, createPostgreSQLClient, createPostgreSQLFlexibleClient } from "../utils/azureClients";
+import { localize } from '../utils/localize';
+
+export class DatabaseNameStep extends AzureWizardPromptStep<IConnectDBWizardContext> {
+
+    public async prompt(context: IConnectDBWizardContext): Promise<void> {
+        const databaseTreeItem = nonNullProp(context, 'databaseConnectionTreeItem');
+        const azureData = nonNullProp(databaseTreeItem, 'azureData');
+        if (azureData.accountKind?.includes('SQL')) {
+            const client = await createCosmosDBClient(context);
+            const listOfDatabases: string[] = [];
+            (await client.sqlResources.listSqlDatabases(nonNullProp(azureData, 'resourceGroup'), azureData.accountName))._response.parsedBody.forEach(db => { if (db) { listOfDatabases.push(nonNullProp(db, 'name')) } });
+            context.databaseName = (await context.ui.showInputBox({
+                placeHolder: "Database Name",
+                prompt: "Provide a database name",
```

Comment:
We should localize these, and since they are used a few times they should also be constants.